From 3cf5e937f5919de4a39ebb04d24794faa5874743 Mon Sep 17 00:00:00 2001
From: Huang Shijie <b32955@freescale.com>
Date: Wed, 2 May 2012 15:48:45 +0800
Subject: ENGR00175578 serial/imx : disable the clock when the uart is not
 used

Disable the clock when the uart port is not used.

Signed-off-by: Huang Shijie <b32955@freescale.com>
---
 drivers/tty/serial/imx.c |    6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/tty/serial/imx.c b/drivers/tty/serial/imx.c
index a2a33fb..9a6e8b6 100644
--- a/drivers/tty/serial/imx.c
+++ b/drivers/tty/serial/imx.c
@@ -932,6 +932,8 @@ static int imx_startup(struct uart_port *port)
 	unsigned long flags, temp;
 	struct tty_struct *tty;
 
+	clk_enable(sport->clk);
+
 #ifndef CONFIG_SERIAL_CORE_CONSOLE
 	imx_setup_ufcr(sport, 0);
 #endif
@@ -1162,6 +1164,7 @@ static void imx_shutdown(struct uart_port *port)
 		writel(temp, sport->port.membase + UCR4);
 	}
 	spin_unlock_irqrestore(&sport->port.lock, flags);
+	clk_disable(sport->clk);
 }
 
 static void
@@ -1708,6 +1711,7 @@ static int serial_imx_probe(struct platform_device *pdev)
 		goto deinit;
 	platform_set_drvdata(pdev, &sport->port);
 
+	clk_disable(sport->clk);
 	return 0;
 deinit:
 	if (pdata && pdata->exit)
@@ -1737,8 +1741,6 @@ static int serial_imx_remove(struct platform_device *pdev)
 		clk_put(sport->clk);
 	}
 
-	clk_disable(sport->clk);
-
 	if (pdata && pdata->exit)
 		pdata->exit(pdev);
 
-- 
1.7.9.5

